# Kinship Analysis

## Kinship: theory ##

Kinship can be defined as the expected fraction of alleles that two individuals got from the same ancestor(s).&nbsp; We say that two individuals share an allele of a SNP *identical-by-descent* or *IBD* if they inherited the same copy of the allele from a common ancestor.&nbsp; IBD-sharing is different from simply carrying the same allele of a gene (known as *identical-by-state* or *IBS*-sharing), which unrelated individuals may do if the allele is common enough in the population.&nbsp; The *degree* $R$ of relationship may be defined as the effective number of meioses separating the relatives through the equation \begin{equation}\frac{1}{2^R}=\frac{1}{2^{R_1}}+\frac{1}{2^{R_2}},\tag{1}\end{equation}in which $R_i$ is the number of meioses separating the relatives through the first relative's $i$<sup>th</sup> parent.&nbsp; For example, sibs are connected by two meioses through two parents, while a parent and child are connected by one meiosis through one parent: both relationships are degree-1.  

The probability that two relatives share an allele IBD is $\frac{1}{2^R}$, as there is a $\frac{1}{2}$ probability that an allele is passed on in any meiosis, and $R$ is the effective number of meioses or steps between the relatives.&nbsp; If $2\times\frac{1}{2^R}$ is the expected number of alleles shared IBD at any given locus, then the fraction of the genome shared by any two relatives is\begin{equation}r=\frac{2\times\frac{1}{2^R}}{2}=\frac{1}{2^R}.\tag{2}\end{equation}

However, genomic sharing can be realized in different ways depending on the probabilities $\pi_0$, $\pi_1$, and $\pi_2$ that individuals share zero, one, or two copies IBD at a locus.&nbsp; The probability that the relatives inherit both both copies IBD, viz.,\begin{equation}\pi_2=P\left(\text{share 2 IBD}\right)=2^2\frac{1}{2^{R_1}2^{R_2}},\tag{3a}\end{equation} is simply the product of the probabilities of sharing through both parents.&nbsp; The probability of sharing exactly one allele IBD,, viz.,\begin{equation}\pi_1=P\left(\text{share 1 IBD}\right)=2\left(\frac{1}{2^{R_1}}+\frac{1}{2^{R_2}}\right)-2^3\frac{1}{2^{R_1}2^{R_2}},\tag{3b}\end{equation} is the got by finding the probability $2\left(\frac{1}{2^{R_1}}+\frac{1}{2^{R_2}}\right)-2^2\frac{1}{2^{R_1}2^{R_2}}$ of sharing at least one allele IBD less the probability $\pi_2$ of sharing two.&nbsp; Finally, the probability of sharing at zero alleles IBD, viz.,\begin{equation}\pi_0=P\left(\text{share 0 IBD}\right)=1-2\frac{1}{2^{R_1}}-2\frac{1}{2^{R_2}}+2^2\frac{1}{2^{R_1}2^{R_2}},\tag{3c}\end{equation}is got by subtracting the probability $\pi_1+\pi_2$ of sharing at least one allele IBD from 1.&nbsp; The coefficients account for the fact that there are $2$ alleles at each locus and $2^2$ that can be shared.

From (3a)&ndash;(3c), the fraction of the genome shared IBD is\begin{equation}r=\frac{2\pi_2+1\pi_1}{2}=\frac{1}{2^{R_1}}+\frac{1}{2^{R_2}}=\frac{1}{2^R}.\tag{4}\end{equation}But the same value of $r$ can obtain from different values of $\pi_1$ and $\pi_2$.&nbsp; For example, full sibs have a 25% probability of sharing two alleles and a 50% chance of sharing one allele at a locus, for a total fraction $r=0.5$ shared.&nbsp; But a parent and child have 0% chance of sharing two alleles and a 100% chance of sharing one, also giving $r=0.5$.&nbsp; Thus, your parent does is not equal your sibling, despite the fact of your sharing equal amounts of your genome with each of them.&nbsp; Put another way, parents cannot pass on their genotypes to their offspring.

### KING ###

KING [@manichaikul_robust_2010] computes both the probability $\pi_0$ that two relatives share 0 alleles IBD as well as the coefficient of relatedness $\phi=\frac{r}{2}$, defined as the probability that two alleles taken one from each relative are IBD at a locus (the maximum probability is $\frac{1}{2}$ because there is a 50% chance that the alleles chosen come from different parents).&nbsp; The idea is to compare the counts $X$ and $Y$ of the alternative alleles which two individuals each have at a genetic locus.&nbsp; If the pair are from a single ancestral population, the expected values and variances of the allele counts are $\mathbb{E}\left(X\right)=\mathbb{E}\left(Y\right)=2p$ and $\sigma_X^2=\mathbb{E}\left(X^2\right)-\mathbb{E}\left(X\right)^2=\mathbb{E}\left(Y^2\right)-\mathbb{E}\left(Y\right)^2=2p\left(1-p\right)$.&nbsp; Thus the expected value of the difference $\mathbb{E}\left(\left(X-Y\right)^2\right)=\mathbb{E}\left(X^2\right)+\mathbb{E}\left(Y^2\right)-2\mathbb{E}\left(XY\right)$ is\begin{equation}\frac{\mathbb{E}\left(\left(X-Y\right)^2\right)}{\sigma_X^2+\sigma_Y^2}=1-\frac{\sigma_{XY}}{\sigma_X\sigma_Y}=1-r,\tag{5}\end{equation}where $\sigma_{XY}=\mathbb{E}\left(XY\right)-\mathbb{E}\left(X\right)\mathbb{E}\left(Y\right)$ is the covariance of the genotype counts and $r_{ij}=2\phi_{ij}$ is the genetic correlation between individuals $i$ and $j$; the latter can be interpreted as the amount of the genome shared IBD.

KING estimates $\phi$ from Eq. (5) by counting the number $N$ of loci at which two individuals are heterozygous $Aa,Aa$ or opposite homozygous $AA,aa$, as well as the total number of alleles at which each individual is heterozygous $Aa$:\begin{equation}\hat{\phi_{ij}}=\frac{N_{Aa,Aa}-2N_{AA,aa}}{N_{Aa}^{\left(i\right)}+N_{Aa}^{\left(j\right)}}.\tag{6}\end{equation}From (6) it can be seen that shared heterozygous sites increase the estimated relatedness, and that unshared homozygous sites decrease relatedness.&nbsp;  Eq. (6) is called a "robust" estimator because it measures relatedness in a purely pairwise fashion: it does not rely on population estimates of allele frequencies.&nbsp; However, if the individuals are not of the same genetic background, the allele frequency $p$ is not well-defined and Eq. (5) does not hold, leading to negative estimates of $\phi$; this feature is not necessarily a problem, as it helps us to distinguish different ancestries within a single population.

### PC-AiR

KING provides an estimate of relatedness.&nbsp; But as cryptic relatedness can skew ancestry estimates determined by PCA, we should use kinship to correct PCA.&nbsp; This is where principal components analysis in related samples, or PC-AiR [@conomos_robust_2015] comes in handy.&nbsp;PC-AiR identifies a subset of $\mathcal{U}$ of individuals who are unrelated to each other (via negative KING kinship estimates), but maximally related to individuals in a remaining set $\mathcal{R}$.&nbsp;Recalling the math in [Week 1](), the matrix\begin{equation}\frac{1}{m}\mathbf{W}=\mathbf{X}^T\mathbf{U}=\mathbf{V}\mathbf{\Sigma},\tag{7}\end{equation}so that $\frac{1}{m}w_{jk}\lambda_{kk}^{-1}=x_{ij}u_{ik}\lambda_{kk}^{-1}=v_{jk}$ is the genotype at SNP $k$ in ancestry $j$ inferred from subjects' genotypes $\mathbb{X}$ who are in subset $\mathcal{U}$.&nbsp; Then if we have an $m\times n_u$ genotype matrix $\mathbf{X}'$ of individuals in the subset $\mathcal{R}$, we can "apply" $\mathbf{W}$ to get an estimate of the "ancestry-corrected" PCs by:\begin{equation}\frac{1}{m}\mathbf{X}'\mathbf{W}\mathbf{\Sigma}^{-1}=\mathbf{X}'\mathbf{V}.\tag{8}\end{equation}Eq. (8) is the matrix $x'_{i\cdot}v_{\cdot j}$ of similarities between subjects $i$ in set $\mathcal{R}$ and ancestry $j$.&nbsp; In this way, we have estimated genotype principal components for all individuals (i.e., those in $\mathcal{U}$ and $\mathcal{R}$) without the problem confounding due to cryptic relatedness.

### PC-Relate

Once we have ancestry-corrected principal components for all individuals, we can obtain "corrected" expected genotypes $\mathbb{E}\left(X_{ik}\right)=2p_{ik}$ of SNPs $k$ for populations of admixed individuals $i$ using a method known as PC-Relate [@conomos_model-free_2016].&nbsp;  Because in populations with both shared genetic ancestry and cryptic relatedness, we can use the PC estimates $v_{kj}$ of the allele frequencies in each population $j$ to express the scaled genotypes $x_{ij}=\frac{g_{ik}-2p_k}{2p_k\left(1-p_k\right)}$ as\begin{equation}\mathbb{E}\left(x_{ik}\mid u_{ij}\right)=u_{ij}\lambda_{jj}v_{kj},\tag{9}\end{equation}or in terms of the measured genotypes:\begin{equation}\mathbb{E}\left(g_{ik}\mid u_{ij}\right)=2p_k+u_{ij}\lambda_{jj}v_{kj}\cdot 2p_k\left(1-p_k\right),\tag{10}\end{equation} in which $p_k$ is the allele frequency estimated from the entire sample.&nbsp; Eq. (10) says that if we make a graph of each individual's genotype $g_{ik}=0,1,2$ at SNP $k$ vs. the (scaled) of the amount the individual's population-$j$ genetic ancestry, the slope $\lambda_{jj}v_{kj}2p_k\left(1-p_k\right)$ should be the expected allele frequency in population $j$.&nbsp; In practice, the slopes are obtained from linear regression of the observed genotypes on genetic PCs, and the updated expected allele frequencies $\mathbb{E}\left(g_{ik}\mid u_{ij}\right)=2p_{ik}$ of SNPs $k$ for individuals $i$ are found by the corresponding value on the best-fit hyperplane, given individual $i$'s genetic ancestry $u_{ij}$.&nbsp; We can then obtain a new genetic relationship matrix:\begin{equation}2\hat{\varphi}_{ij}=\frac{\sum_k\left(g_{ik}-2p_{ik}\right)\left(g_{jk}-2p_{jk}\right)}{2\sqrt{p_{ik}\left(1-p_{ik}\right)p_{jk}\left(1-p_{jk}\right)}}.\tag{11}\end{equation}This new GRM will be used in [Week 3]() when we attempt to fit a linear mixed model for the effect of genotype on disease risk; for now we will use it to make an updated table of the relationship between individuals.

## Kinship: practice

Now we'll try kinship analysis for ourselves using a simulated GWAS dataset.  This dataset was created using the program bioGWAS [@changalidis_biogwas_2023], which uses other programs to simulate haplotypes [@su_hapgen2_2011] and phenotypes [@meyer_phenotypesimulator_2018] based on a set of input variants previously associated with breast cancer [@michailidou_genome-wide_2015] and made available by the MRC Interactive Epidemiology Unit's Open GWAS Project [@elsworth_mrc_2020].  This particular [vcf file](https://github.com/wletsou/BIOL-350/raw/master/docs/EUR_BCa.vcf.gz) contains 1000 chromosome 10 genotypes at 345,130 variants (304,770 of which are SNPs) generated from 263 1KGP females of European origin.  Download and unzip the file:

```
gunzip EUR_BCa.vcf.gz
```

and move it to your desired location.  The unzipped file is readable in plain text format, although it is very big.

Next load the following R packages:

```
library(GWASTools)
library(SNPRelate)
library(GENESIS)
```


# References

<div id="refs"></div>