% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{book}
\usepackage{amsmath,amssymb}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math} % this also loads fontspec
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
\usepackage{lmodern}
\ifPDFTeX\else
  % xetex/luatex font selection
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\usepackage{longtable,booktabs,array}
\usepackage{calc} % for calculating minipage widths
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi
\usepackage[style=apalike,]{biblatex}
\addbibresource{book.bib}
\addbibresource{packages.bib}
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same}
\hypersetup{
  pdftitle={BIOL 350: Bioinformatics},
  pdfauthor={William Letsou},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}

\title{BIOL 350: Bioinformatics}
\author{William Letsou}
\date{2024-03-21}

\begin{document}
\maketitle

{
\setcounter{tocdepth}{1}
\tableofcontents
}
\hypertarget{intro}{%
\chapter{Introduction}\label{intro}}

Welcome to BIOL-350 (Spring 2024) at New York Tech!~ For this set of lectures, follow along with the slides \href{https://wletsou.github.io/BIOL-350/Biol\%20350\%20slides.pdf}{here}.~ Each chapter describes the theory and practice of the step in an analysis pipeline for you to conduct your own genetic association study.

\hypertarget{principal-components-analysis}{%
\chapter{Principal Components Analysis}\label{principal-components-analysis}}

This week we'll see how individuals can be separated by genetic ancestry using principal components analysis.~ We'll practice applying PCA on a subset of the 1KGP data.

\hypertarget{pricipal-components-analysis-theory}{%
\section{Pricipal components analysis: theory}\label{pricipal-components-analysis-theory}}

The populations in our dataset can be separated into clusters based on their genotypes.~ The inferred groups help control for confounding due to ancestry, and are also more reliable than self-reported race in association studies.~ To see how it works, suppose \(\mathbf{X}\) is an \(n\times m\) (standardized) genotype matrix with individuals down the rows and SNPs across the columns.~ Principal components analysis (PCA) says we can find an \(m\times n\) matrix \(\mathbf{V}\), a diagonal \(n\times n\) matrix \(\mathbf{\Sigma}\), and an \(n\times n\) matrix \(\mathbf{U}\) satisfying \begin{equation}\mathbf{X}=\mathbf{U}\mathbf{\Sigma}\mathbf{V}^T.\tag{1}\end{equation} If we think of \(\mathbf{V}\) as the the (standardized) SNP genotypes of an ``ideal'' person of a certain ancestry, then \begin{equation}x_{j\cdot} v_{\cdot i}=u_{ji}\lambda_i\end{equation} represents the amount of idealized person \(i\) in actual person \(j\), up to some proportionality constant \(\lambda_i\) that depends on the individual.~ Then rows \(j\) of \(\mathbf{U}\) are the \emph{ancestries} of person \(j\), and columns \(i\) of \(\mathbf{U}\) are the ancestries of each individual on ancestry \(i\).~ It is important to remember that these ``idealized'' ancestries do not necessarily correspond with our preconceived notions of ancestry, so we cannot interpret them as ``European,'' ``African,'' or ``Asian,'' say.~ If we rearrange Eq. (1) and use the fact that the columns of \(\mathbf{U}\) and \(\mathbf{V}\) are \emph{orthonormal}, we can find that \begin{equation}\mathbf{X}\mathbf{X}^T\mathbf{U}=\mathbf{U}\mathbf{\Sigma}^2,\end{equation} meaning that the columns of \(\mathbf{U}\) are the eigenvectors of the matrix \begin{equation}\frac{1}{m}\mathbf{X}\mathbf{X}^T\tag{2}\end{equation} whose \(\left(i,j\right)\) entry is the genetic correlation between individuals \(i\) and \(j\), sometimes known as the \emph{genomic relationship matrix} or GRM.~ PCA works by finding the first several eigenvectors of the GRM and plotting each individual's ancestry along each orthogonal vector in a rectangular grid.~ Clusters of individuals in this grid represent distinct ancestry groups.

\hypertarget{principal-components-analysis-practice}{%
\section{Principal components analysis: practice}\label{principal-components-analysis-practice}}

\hypertarget{importing-the-data}{%
\subsection{Importing the data}\label{importing-the-data}}

To do PCA in R, we'll need to load the libraries SNPRelate and SeqArray:

\begin{verbatim}
library(SNPRelate)
library(SeqArray)
\end{verbatim}

Download the \href{https://github.com/wletsou/BIOL-350/raw/master/docs/CHB\%2BYRI\%2BCEU.chr1.vcf.gz}{chr1 vcf file} containing just the CEU, YRI, and CHB populations.~ Once you have the file, store its name as a variable:

\begin{verbatim}
vcf <- "path/to/file.vcf.gz"
\end{verbatim}

Now we'll convert the vcf format to gds format, retaining the base filename and changing the vcf.gz extension to gds.~ (This may take a minute to complete.)~ Then we'll import the gds file as a gds object.

\begin{verbatim}
seqVCF2GDS(vcf.fn = vcf,"path/to/file.gds") # convert vcf to gds with a new file name
genofile <- seqOpen("path/to/file.gds") # import the gds object
\end{verbatim}

You can can see the various fields under genofile by printing it.~ To access the data in one of the fields, do

\begin{verbatim}
seqGetData(genofile,"sample.id") # view the sample ids
\end{verbatim}

where the name of the field is enclosed in quotes.

\hypertarget{running-pca}{%
\subsection{Running PCA}\label{running-pca}}

To run PCA in R, simply do

\begin{verbatim}
pca <- snpgdsPCA(genofile) # runs PCA
\end{verbatim}

to create an objects with 32 eignevectors of the GRM.~ Make a data frame of the first several eigenvectors along with subject ids:

\begin{verbatim}
df.pca <- data.frame(sample = pca$sample.id,EV1 = pca$eigenvect[,1],EV2 = pca$eigenvect[,2],EV3 = pca$eigenvect[,3],stringsAsFactors = FALSE)
\end{verbatim}

We'll plot individuals along EV1, EV2, and EV3 in several two-dimensional projections.~ But we'll want to see how the clustering done by PCA corresponds to individuals' self-reported race; for that we'll need another column in our data frame.

\hypertarget{getting-population-labels}{%
\subsection{Getting population labels}\label{getting-population-labels}}

The \href{https://raw.githubusercontent.com/wletsou/BIOL-350/master/docs/CHB\%2BYRI\%2BCEU.txt}{indivs file} contains each subject id along with its 1KGP population group.~ Let's import it now:

\begin{verbatim}
indivs <- read.table("path/to/CHB+YRI+CEU.txt",header = FALSE)
colnames(indivs) <- c("id","pop")
\end{verbatim}

The second field of this table is pop, an assignment to each id of one of three 1KG population groups.~ We want to match the right ID in indivs to the right ID in df.pca so that we can color our PCA plots by population.~ If the tables are in the same order, matching will be easy, but it not, we have to use the match(x,y) function, which finds the positions in y corresponding to the same items in x.~ Thus we can make a new column pop in df.pca with the corresponding pop values from indivs by

\begin{verbatim}
df.pca$pop[match(indivs$id,df.pca$sample)] <- indivs$pop # find the population group of each individual in df.pca
\end{verbatim}

\hypertarget{plotting}{%
\subsection{Plotting}\label{plotting}}

Now that we have a column of population labels, we can make a scatter plot colored by treating the pop column as vector of factors; we can get the unique values of a factor vector by applying the function levels() to it.~ A plot of the second principal component vs.~the first can then be generated by

\begin{verbatim}
par(mar = c(5.1,5.1,4.1,2.1) ) 
plot(df.pca$EV1,df.pca$EV2,pch = 19,col = factor(df.pca$pop),xlab = "PC1",ylab = "PC2",cex.lab = 1.5,cex.axis = 1.5,cex.main = 1.5) # plot of PC2 vs. PC1
legend("topright",legend = levels(factor(df.pca$pop)),bty = "n",pch = 19,col = factor(levels(factor(df.pca$pop))),pt.cex = 1,cex = 1.5,x.intersp = 0.2) # with a legend
\end{verbatim}

Move the legend around if it covers any points, and make similar plots for the other two comparisons between the first three PCs.

Finally, close the connection to the gds file when you are done:

\begin{verbatim}
seqClose(genofile)
\end{verbatim}

\hypertarget{to-turn-in}{%
\section{To turn in:}\label{to-turn-in}}

Make three (nicely formatted) plots of:

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  PC2 vs.~PC1
\item
  PC3 vs.~PC1
\item
  PC3 vs.~PC2
\end{enumerate}

For each plot, discuss:

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  Whether the populations appear to be well separated in PCA space
\item
  What the gradients of the different PCs represent, that is, what axis of variation each PC appears to explain
\item
  How to subset your df.pca data frame to isolate individuals of each population (i.e., provide code)
\end{enumerate}

\hypertarget{kinship-analysis}{%
\chapter{Kinship Analysis}\label{kinship-analysis}}

\hypertarget{kinship-theory}{%
\section{Kinship: theory}\label{kinship-theory}}

Kinship can be defined as the expected fraction of alleles that two individuals got from the same ancestor(s).~ We say that two individuals share an allele of a SNP \emph{identical-by-descent} or \emph{IBD} if they inherited the same copy of the allele from a common ancestor.~ IBD-sharing is different from simply carrying the same allele of a gene (known as \emph{identical-by-state} or \emph{IBS}-sharing), which unrelated individuals may do if the allele is common enough in the population.~ The \emph{degree} \(R\) of relationship may be defined as the effective number of meioses separating the relatives through the equation \begin{equation}\frac{1}{2^R}=\frac{1}{2^{R_1}}+\frac{1}{2^{R_2}},\tag{1}\end{equation}in which \(R_i\) is the number of meioses separating the relatives through the first relative's \(i\)th parent.~ For example, sibs are connected by two meioses through two parents, while a parent and child are connected by one meiosis through one parent: both relationships are degree-1.

The probability that two relatives share an allele IBD is \(\frac{1}{2^R}\), as there is a \(\frac{1}{2}\) probability that an allele is passed on in any meiosis, and \(R\) is the effective number of meioses or steps between the relatives.~ If \(2\times\frac{1}{2^R}\) is the expected number of alleles shared IBD at any given locus, then the fraction of the genome shared by any two relatives is\begin{equation}r=\frac{2\times\frac{1}{2^R}}{2}=\frac{1}{2^R}.\tag{2}\end{equation}

However, genomic sharing can be realized in different ways depending on the probabilities \(\pi_0\), \(\pi_1\), and \(\pi_2\) that individuals share zero, one, or two copies IBD at a locus.~ The probability that the relatives inherit both both copies IBD, viz.,\begin{equation}\pi_2=P\left(\text{share 2 IBD}\right)=2^2\frac{1}{2^{R_1}2^{R_2}},\tag{3a}\end{equation} is simply the product of the probabilities of sharing through both parents.~ The probability of sharing exactly one allele IBD,, viz.,\begin{equation}\pi_1=P\left(\text{share 1 IBD}\right)=2\left(\frac{1}{2^{R_1}}+\frac{1}{2^{R_2}}\right)-2^3\frac{1}{2^{R_1}2^{R_2}},\tag{3b}\end{equation} is the got by finding the probability \(2\left(\frac{1}{2^{R_1}}+\frac{1}{2^{R_2}}\right)-2^2\frac{1}{2^{R_1}2^{R_2}}\) of sharing at least one allele IBD less the probability \(\pi_2\) of sharing two.~ Finally, the probability of sharing at zero alleles IBD, viz.,\begin{equation}\pi_0=P\left(\text{share 0 IBD}\right)=1-2\frac{1}{2^{R_1}}-2\frac{1}{2^{R_2}}+2^2\frac{1}{2^{R_1}2^{R_2}},\tag{3c}\end{equation}is got by subtracting the probability \(\pi_1+\pi_2\) of sharing at least one allele IBD from 1.~ The coefficients account for the fact that there are \(2\) alleles at each locus and \(2^2\) that can be shared.

From (3a)--(3c), the fraction of the genome shared IBD is\begin{equation}r=\frac{2\pi_2+1\pi_1}{2}=\frac{1}{2^{R_1}}+\frac{1}{2^{R_2}}=\frac{1}{2^R}.\tag{4}\end{equation}But the same value of \(r\) can obtain from different values of \(\pi_1\) and \(\pi_2\).~ For example, full sibs have a 25\% probability of sharing two alleles and a 50\% chance of sharing one allele at a locus, for a total fraction \(r=0.5\) shared.~ But a parent and child have 0\% chance of sharing two alleles and a 100\% chance of sharing one, also giving \(r=0.5\).~ Thus, your parent does is not equal your sibling, despite the fact of your sharing equal amounts of your genome with each of them.~ Put another way, parents cannot pass on their genotypes to their offspring.

\hypertarget{king}{%
\section{KING}\label{king}}

KING\cite{manichaikul_robust_2010} computes both the probability \(\pi_0\) that two relatives share 0 alleles IBD as well as the coefficient of relatedness \(\phi=\frac{r}{2}\), defined as the probability that two alleles taken one from each relative are IBD at a locus (the maximum probability is \(\frac{1}{2}\) because there is a 50\% chance that the alleles chosen come from different parents).~ The idea is to compare the counts \(X\) and \(Y\) of the alternative alleles which two individuals each have at a genetic locus.~ If pair are from a single ancestral population, the expected values and variances of the allele counts are \(\mathbb{E}\left(X\right)=\mathbb{E}\left(Y\right)=2p\) and \(\sigma_X^2=\mathbb{E}\left(X^2\right)-\mathbb{E}\left(X\right)^2=\mathbb{E}\left(Y^2\right)-\mathbb{E}\left(Y\right)^2=2p\left(1-p\right)\).~ Thus the expected value of the difference \(\mathbb{E}\left(X^2-Y^2\right)=\mathbb{E}\left(X^2\right)+\mathbb{E}\left(Y^2\right)-2\mathbb{E}\left(XY\right)\) is\begin{equation}\frac{\mathbb{E}\left(X^2-Y^2\right)}{\sigma_X^2+\sigma_Y^2}=1-\frac{\sigma_{XY}}{\sigma_X\sigma_Y}=1-r,\tag{5}\end{equation}where \(\sigma_{XY}=\mathbb{E}\left(XY\right)-\mathbb{E}\left(X\right)\mathbb{E}\left(Y\right)\) is the covariance of the genotype counts and \(r=2\phi\) is the genetic correlation between two individuals; the latter can be interprettted as the amount of the genome shared IBD.

KING estimates \(\phi\) from Eq. (5) by counting the number \(N\) of loci at which two individuals are heterozygous \(Aa,Aa\) or opposite homozygous \(AA,aa\), as well as the total number of alleles at which each individual is heterozygous \(Aa\):\begin{equation}\hat{\phi_{ij}}=\frac{N_{Aa,Aa}-2N_{AA,aa}}{N_{Aa}^{\left(i\right)}+N_{Aa}^{\left(j\right)}}.\tag{6}\end{equation}From (6) it can be seen that shared heterozygous sites increase the estimated relatedness, and that unshared homozygous sites decrease relatedness.~ Eq. (6) is called a ``robust'' estimator because it measures relatedness in a purely pairwise fashion: it does not rely on population estimates of allele frequencies.~ However, if the individuals are not of the same genetic background, the allele frequency \(p\) is not well-defined and Eq. (5) does not hold, leading to negative estimates of \(\phi\); this feature is not necessarily a problem, as it helps us to distinguish different ancestries within a single population.

\printbibliography

\end{document}
