<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 References | BIOL 350: Bioinformatics</title>
  <meta name="description" content="Tutorial on genetic association studies." />
  <meta name="generator" content="bookdown 0.38 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 References | BIOL 350: Bioinformatics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Tutorial on genetic association studies." />
  <meta name="github-repo" content="openscapes/series" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 References | BIOL 350: Bioinformatics" />
  
  <meta name="twitter:description" content="Tutorial on genetic association studies." />
  

<meta name="author" content="William Letsou" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="Week3.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="lib/css/bootstrap.min.css" type="text/css" />
<link rel="stylesheet" href="lib/css/style.css" type="text/css" />
<link rel="stylesheet" href="lib/css/lesson.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">BIOL 350: Bioinformatics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="Week1.html"><a href="Week1.html"><i class="fa fa-check"></i><b>2</b> Principal Components Analysis</a>
<ul>
<li class="chapter" data-level="2.1" data-path="Week1.html"><a href="Week1.html#Week1_theory"><i class="fa fa-check"></i><b>2.1</b> Pricipal components analysis: theory</a></li>
<li class="chapter" data-level="2.2" data-path="Week1.html"><a href="Week1.html#Week1_practice"><i class="fa fa-check"></i><b>2.2</b> Principal components analysis: practice</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="Week1.html"><a href="Week1.html#Week1_practice_import"><i class="fa fa-check"></i><b>2.2.1</b> Importing the data</a></li>
<li class="chapter" data-level="2.2.2" data-path="Week1.html"><a href="Week1.html#Week1_practice_pca"><i class="fa fa-check"></i><b>2.2.2</b> Running PCA</a></li>
<li class="chapter" data-level="2.2.3" data-path="Week1.html"><a href="Week1.html#Week1_practice_labels"><i class="fa fa-check"></i><b>2.2.3</b> Getting the population labels</a></li>
<li class="chapter" data-level="2.2.4" data-path="Week1.html"><a href="Week1.html#Week1_practice_plot"><i class="fa fa-check"></i><b>2.2.4</b> Plotting</a></li>
<li class="chapter" data-level="2.2.5" data-path="Week1.html"><a href="Week1.html#Week1_practice_turnin"><i class="fa fa-check"></i><b>2.2.5</b> To turn in:</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="Week1.html"><a href="Week1.html#Week1_ref"><i class="fa fa-check"></i><b>2.3</b> References</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="Week2.html"><a href="Week2.html"><i class="fa fa-check"></i><b>3</b> Kinship Analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="Week2.html"><a href="Week2.html#Week2_theory"><i class="fa fa-check"></i><b>3.1</b> Kinship: theory</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="Week2.html"><a href="Week2.html#Week2_theory_king"><i class="fa fa-check"></i><b>3.1.1</b> KING</a></li>
<li class="chapter" data-level="3.1.2" data-path="Week2.html"><a href="Week2.html#Week2_theory_pcair"><i class="fa fa-check"></i><b>3.1.2</b> PC-AiR</a></li>
<li class="chapter" data-level="3.1.3" data-path="Week2.html"><a href="Week2.html#Week2_theory_pcrelate"><i class="fa fa-check"></i><b>3.1.3</b> PC-Relate</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="Week2.html"><a href="Week2.html#Week2_practice"><i class="fa fa-check"></i><b>3.2</b> Kinship: practice</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="Week2.html"><a href="Week2.html#Week2_practice_import"><i class="fa fa-check"></i><b>3.2.1</b> Importing data</a></li>
<li class="chapter" data-level="3.2.2" data-path="Week2.html"><a href="Week2.html#Week2_practice_pruning"><i class="fa fa-check"></i><b>3.2.2</b> PCA and LD-pruning</a></li>
<li class="chapter" data-level="3.2.3" data-path="Week2.html"><a href="Week2.html#Week2_practice_king"><i class="fa fa-check"></i><b>3.2.3</b> KING IBD and kinship calculation</a></li>
<li class="chapter" data-level="3.2.4" data-path="Week2.html"><a href="Week2.html#Week2_practice_pcair"><i class="fa fa-check"></i><b>3.2.4</b> PC-AiR</a></li>
<li class="chapter" data-level="3.2.5" data-path="Week2.html"><a href="Week2.html#Week2_practice_pcrelate"><i class="fa fa-check"></i><b>3.2.5</b> PC-Relate</a></li>
<li class="chapter" data-level="3.2.6" data-path="Week2.html"><a href="Week2.html#Week2_practice_turnin"><i class="fa fa-check"></i><b>3.2.6</b> To turn in:</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="Week2.html"><a href="Week2.html#Week2_ref"><i class="fa fa-check"></i><b>3.3</b> References</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="Week3.html"><a href="Week3.html"><i class="fa fa-check"></i><b>4</b> Association testing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="Week3.html"><a href="Week3.html#Week3_theory"><i class="fa fa-check"></i><b>4.1</b> Statistics</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="Week3.html"><a href="Week3.html#Week3_theory_or"><i class="fa fa-check"></i><b>4.1.1</b> The odds ratio</a></li>
<li class="chapter" data-level="4.1.2" data-path="Week3.html"><a href="Week3.html#Week3_theory_logit"><i class="fa fa-check"></i><b>4.1.2</b> The logistic model of disease risk</a></li>
<li class="chapter" data-level="4.1.3" data-path="Week3.html"><a href="Week3.html#Week3_theory_glm"><i class="fa fa-check"></i><b>4.1.3</b> Linear mixed models</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="Week3.html"><a href="Week3.html#Week3_practice"><i class="fa fa-check"></i><b>4.2</b> Practice</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="Week3.html"><a href="Week3.html#Week3_practice_pheno"><i class="fa fa-check"></i><b>4.2.1</b> Get phenotype</a></li>
<li class="chapter" data-level="4.2.2" data-path="Week3.html"><a href="Week3.html#Week3_practice_fit"><i class="fa fa-check"></i><b>4.2.2</b> Fitting the model</a></li>
<li class="chapter" data-level="4.2.3" data-path="Week3.html"><a href="Week3.html#Week3_practice_analysis"><i class="fa fa-check"></i><b>4.2.3</b> Analyzing the results</a></li>
<li class="chapter" data-level="4.2.4" data-path="Week3.html"><a href="Week3.html#Week3_practice_turnin"><i class="fa fa-check"></i><b>4.2.4</b> To turn in:</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="Week3.html"><a href="Week3.html#Week3_ref"><i class="fa fa-check"></i><b>4.3</b> References</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="refs.html"><a href="refs.html"><i class="fa fa-check"></i><b>5</b> References</a></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">BIOL 350: Bioinformatics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="refs" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> References<a href="refs.html#refs" class="anchor-section" aria-label="Anchor link to header"></a></h1>

<div id="refs" class="references csl-bib-body">
<div class="csl-entry">
1. The 1000 Genomes Project Consortium, Auton A, Abecasis GR, Steering committee, Altshuler DM, Durbin RM, et al. <a href="https://doi.org/10.1038/nature15393">A global reference for human genetic variation</a>. Nature. 2015;526:68–74.
</div>
<div class="csl-entry">
2. Price AL, Patterson NJ, Plenge RM, Weinblatt ME, Shadick NA, Reich D. <a href="https://doi.org/10.1038/ng1847">Principal components analysis corrects for stratification in genome-wide association studies</a>. Nature Genetics. 2006;38:904–9.
</div>
<div class="csl-entry">
3. Shlens J. <a href="http://arxiv.org/abs/1404.1100">A <span>Tutorial</span> on <span>Principal</span> <span>Component</span> <span>Analysis</span></a>. 2014.
</div>
<div class="csl-entry">
4. Zheng X, Levine D, Shen J, Gogarten SM, Laurie C, Weir BS. <a href="https://doi.org/10.1093/bioinformatics/bts606">A high-performance computing toolset for relatedness and principal component analysis of <span>SNP</span> data</a>. Bioinformatics. 2012;28:3326–8.
</div>
<div class="csl-entry">
5. Zheng X, Gogarten SM, Lawrence M, Stilp A, Conomos MP, Weir BS, et al. <a href="https://doi.org/10.1093/bioinformatics/btx145"><span>SeqArray</span>—a storage-efficient high-performance data format for <span>WGS</span> variant calls</a>. Bioinformatics. 2017;33:2251–7.
</div>
<div class="csl-entry">
6. Manichaikul A, Mychaleckyj JC, Rich SS, Daly K, Sale M, Chen W-M. <a href="https://doi.org/10.1093/bioinformatics/btq559">Robust relationship inference in genome-wide association studies</a>. Bioinformatics. 2010;26:2867–73.
</div>
<div class="csl-entry">
7. Conomos MP, Miller MB, Thornton TA. <a href="https://doi.org/10.1002/gepi.21896">Robust <span>Inference</span> of <span>Population</span> <span>Structure</span> for <span>Ancestry</span> <span>Prediction</span> and <span>Correction</span> of <span>Stratification</span> in the <span>Presence</span> of <span>Relatedness</span></a>. Genetic Epidemiology. 2015;39:276–93.
</div>
<div class="csl-entry">
8. Conomos MP, Reiner AP, Weir BS, Thornton TA. <a href="https://doi.org/10.1016/j.ajhg.2015.11.022">Model-free <span>Estimation</span> of <span>Recent</span> <span>Genetic</span> <span>Relatedness</span></a>. The American Journal of Human Genetics. 2016;98:127–48.
</div>
<div class="csl-entry">
9. Changalidis AI, Alexeev DA, Nasykhova YA, Glotov AS, Barbitoff YA. <a href="https://doi.org/10.3390/biology13010010"><span class="nocase">bioGWAS</span>: <span>A</span> <span>Simple</span> and <span>Flexible</span> <span>Tool</span> for <span>Simulating</span> <span>GWAS</span> <span>Datasets</span></a>. Biology. 2023;13:10.
</div>
<div class="csl-entry">
10. Su Z, Marchini J, Donnelly P. <a href="https://doi.org/10.1093/bioinformatics/btr341"><span>HAPGEN2</span>: Simulation of multiple disease <span>SNPs</span></a>. Bioinformatics. 2011;27:2304–5.
</div>
<div class="csl-entry">
11. Meyer HV, Birney E. <a href="https://doi.org/10.1093/bioinformatics/bty197"><span>PhenotypeSimulator</span>: <span>A</span> comprehensive framework for simulating multi-trait, multi-locus genotype to phenotype relationships</a>. Bioinformatics. 2018;34:2951–6.
</div>
<div class="csl-entry">
12. Michailidou K, BOCS, kConFab Investigators, AOCS Group, NBCS, GENICA Network, et al. <a href="https://doi.org/10.1038/ng.3242">Genome-wide association analysis of more than 120,000 individuals identifies 15 new susceptibility loci for breast cancer</a>. Nature Genetics. 2015;47:373–80.
</div>
<div class="csl-entry">
13. Elsworth B, Lyon M, Alexander T, Liu Y, Matthews P, Hallett J, et al. <a href="https://doi.org/10.1101/2020.08.10.244293">The <span>MRC</span> <span>IEU</span> <span>OpenGWAS</span> data infrastructure</a>. 2020.
</div>
<div class="csl-entry">
14. Gogarten SM, Sofer T, Chen H, Yu C, Brody JA, Thornton TA, et al. <a href="https://doi.org/10.1093/bioinformatics/btz567">Genetic association testing using the <span>GENESIS</span> <span>R</span>/<span>Bioconductor</span> package</a>. Bioinformatics. 2019;35:5346–8.
</div>
<div class="csl-entry">
15. Henderson CR, Kempthorne O, Searle SR, Von Krosigk CM. <a href="https://doi.org/10.2307/2527669">The <span>Estimation</span> of <span>Environmental</span> and <span>Genetic</span> <span>Trends</span> from <span>Records</span> <span>Subject</span> to <span>Culling</span></a>. Biometrics. 1959;15:192.
</div>
<div class="csl-entry">
16. Patterson HD, Thompson R. <a href="https://doi.org/10.1093/biomet/58.3.545">Recovery of inter-block information when block sizes are unequal</a>. Biometrika. 1971;58:545–54.
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="Week3.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/wletsou/BIOL 350/edit/master/04-Bibliography.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["series.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
},
"citation_package": "biblatex"
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
