<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Principal Components Analysis | BIOL 350: Bioinformatics</title>
  <meta name="description" content="Tutorial on genetic association studies." />
  <meta name="generator" content="bookdown 0.38 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Principal Components Analysis | BIOL 350: Bioinformatics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Tutorial on genetic association studies." />
  <meta name="github-repo" content="openscapes/series" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Principal Components Analysis | BIOL 350: Bioinformatics" />
  
  <meta name="twitter:description" content="Tutorial on genetic association studies." />
  

<meta name="author" content="William Letsou" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="Week2.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="lib/css/bootstrap.min.css" type="text/css" />
<link rel="stylesheet" href="lib/css/style.css" type="text/css" />
<link rel="stylesheet" href="lib/css/lesson.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">BIOL 350: Bioinformatics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="Week1.html"><a href="Week1.html"><i class="fa fa-check"></i><b>2</b> Principal Components Analysis</a>
<ul>
<li class="chapter" data-level="2.1" data-path="Week1.html"><a href="Week1.html#Week1_theory"><i class="fa fa-check"></i><b>2.1</b> Pricipal components analysis: theory</a></li>
<li class="chapter" data-level="2.2" data-path="Week1.html"><a href="Week1.html#Week1_practice"><i class="fa fa-check"></i><b>2.2</b> Principal components analysis: practice</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="Week1.html"><a href="Week1.html#Week1_practice_import"><i class="fa fa-check"></i><b>2.2.1</b> Importing the data</a></li>
<li class="chapter" data-level="2.2.2" data-path="Week1.html"><a href="Week1.html#Week1_practice_pca"><i class="fa fa-check"></i><b>2.2.2</b> Running PCA</a></li>
<li class="chapter" data-level="2.2.3" data-path="Week1.html"><a href="Week1.html#Week1_practice_labels"><i class="fa fa-check"></i><b>2.2.3</b> Getting the population labels</a></li>
<li class="chapter" data-level="2.2.4" data-path="Week1.html"><a href="Week1.html#Week1_practice_plot"><i class="fa fa-check"></i><b>2.2.4</b> Plotting</a></li>
<li class="chapter" data-level="2.2.5" data-path="Week1.html"><a href="Week1.html#Week1_practice_turnin"><i class="fa fa-check"></i><b>2.2.5</b> To turn in:</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="Week2.html"><a href="Week2.html"><i class="fa fa-check"></i><b>3</b> Kinship Analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="Week2.html"><a href="Week2.html#Week2_theory"><i class="fa fa-check"></i><b>3.1</b> Kinship: theory</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="Week2.html"><a href="Week2.html#Week2_theory_king"><i class="fa fa-check"></i><b>3.1.1</b> KING</a></li>
<li class="chapter" data-level="3.1.2" data-path="Week2.html"><a href="Week2.html#Week2_theory_pcair"><i class="fa fa-check"></i><b>3.1.2</b> PC-AiR</a></li>
<li class="chapter" data-level="3.1.3" data-path="Week2.html"><a href="Week2.html#Week2_theory_pcrelate"><i class="fa fa-check"></i><b>3.1.3</b> PC-Relate</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="Week2.html"><a href="Week2.html#Week2_practice"><i class="fa fa-check"></i><b>3.2</b> Kinship: practice</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="Week2.html"><a href="Week2.html#Week2_practice_import"><i class="fa fa-check"></i><b>3.2.1</b> Importing data</a></li>
<li class="chapter" data-level="3.2.2" data-path="Week2.html"><a href="Week2.html#Week2_practice_pruning"><i class="fa fa-check"></i><b>3.2.2</b> PCA and LD-pruning</a></li>
<li class="chapter" data-level="3.2.3" data-path="Week2.html"><a href="Week2.html#Week2_practice_king"><i class="fa fa-check"></i><b>3.2.3</b> KING IBD and kinship calculation</a></li>
<li class="chapter" data-level="3.2.4" data-path="Week2.html"><a href="Week2.html#Week2_practice_pcair"><i class="fa fa-check"></i><b>3.2.4</b> PC-AiR</a></li>
<li class="chapter" data-level="3.2.5" data-path="Week2.html"><a href="Week2.html#Week2_practice_pcrelate"><i class="fa fa-check"></i><b>3.2.5</b> PC-Relate</a></li>
<li class="chapter" data-level="3.2.6" data-path="Week2.html"><a href="Week2.html#Week2_practice_turnin"><i class="fa fa-check"></i><b>3.2.6</b> To turn in:</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="Week2.html"><a href="Week2.html#Week2_ref"><i class="fa fa-check"></i><b>3.3</b> References</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="Week3.html"><a href="Week3.html"><i class="fa fa-check"></i><b>4</b> Association testing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="Week3.html"><a href="Week3.html#Week3_theory"><i class="fa fa-check"></i><b>4.1</b> Statistics</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="Week3.html"><a href="Week3.html#Week3_theory_or"><i class="fa fa-check"></i><b>4.1.1</b> The odds ratio</a></li>
<li class="chapter" data-level="4.1.2" data-path="Week3.html"><a href="Week3.html#Week3_theory_logit"><i class="fa fa-check"></i><b>4.1.2</b> The logistic model of disease risk</a></li>
<li class="chapter" data-level="4.1.3" data-path="Week3.html"><a href="Week3.html#Week3_theory_glm"><i class="fa fa-check"></i><b>4.1.3</b> Linear mixed models</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="Week3.html"><a href="Week3.html#Week3_practice"><i class="fa fa-check"></i><b>4.2</b> Practice</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="Week3.html"><a href="Week3.html#Week3_practice_pheno"><i class="fa fa-check"></i><b>4.2.1</b> Get phenotype</a></li>
<li class="chapter" data-level="4.2.2" data-path="Week3.html"><a href="Week3.html#Week3_practice_fit"><i class="fa fa-check"></i><b>4.2.2</b> Fitting the model</a></li>
<li class="chapter" data-level="4.2.3" data-path="Week3.html"><a href="Week3.html#Week3_practice_analysis"><i class="fa fa-check"></i><b>4.2.3</b> Analyzing the results</a></li>
<li class="chapter" data-level="4.2.4" data-path="Week3.html"><a href="Week3.html#to-turn-in-week3_practice_turnin"><i class="fa fa-check"></i><b>4.2.4</b> To turn in {#Week3_practice_turnin}:</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="Week3.html"><a href="Week3.html#Week3_ref"><i class="fa fa-check"></i><b>4.3</b> References</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">BIOL 350: Bioinformatics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="Week1" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Principal Components Analysis<a href="Week1.html#Week1" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This week we’ll see how individuals can be separated by genetic ancestry using principal components analysis.  We’ll practice applying PCA on a subset of the 1KGP data.</p>
<div id="Week1_theory" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Pricipal components analysis: theory<a href="Week1.html#Week1_theory" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The populations in our dataset can be separated into clusters based on their genotypes.  The inferred groups help control for confounding due to ancestry, and are also more reliable than self-reported race in association studies.  To see how it works, suppose <span class="math inline">\(\mathbf{X}\)</span> is an <span class="math inline">\(n\times m\)</span> (standardized) genotype matrix with individuals down the rows and SNPs across the columns.  Principal components analysis (PCA) says we can find an <span class="math inline">\(m\times n\)</span> matrix <span class="math inline">\(\mathbf{V}\)</span>, a diagonal <span class="math inline">\(n\times n\)</span> matrix <span class="math inline">\(\mathbf{\Sigma}\)</span>, and an <span class="math inline">\(n\times n\)</span> matrix <span class="math inline">\(\mathbf{U}\)</span> satisfying
<span class="math display" id="eq:1-1">\[\begin{equation}
\mathbf{X}=\mathbf{U}\mathbf{\Sigma}\mathbf{V}^T.
\tag{2.1}
\end{equation}\]</span>If we think of <span class="math inline">\(\mathbf{V}\)</span> as the the (standardized) SNP genotypes of an “ideal” person of a certain ancestry, then
<span class="math display" id="eq:1-2">\[\begin{equation}
x_{j\cdot} v_{\cdot i}=u_{ji}\lambda_{jj}
\tag{2.2}
\end{equation}\]</span>represents the amount of idealized person <span class="math inline">\(i\)</span> in actual person <span class="math inline">\(j\)</span>, up to some proportionality constant <span class="math inline">\(\lambda_{jj}\)</span> that depends on the ancestry.  Then rows <span class="math inline">\(j\)</span> of <span class="math inline">\(\mathbf{U}\)</span> are the <em>ancestries</em> of person <span class="math inline">\(j\)</span>, and columns <span class="math inline">\(i\)</span> of <span class="math inline">\(\mathbf{U}\)</span> are the ancestries of each individual on ancestry <span class="math inline">\(i\)</span>.  It is important to remember that these “idealized” ancestries do not necessarily correspond with our preconceived notions of ancestry, so we cannot interpret them as “European,” “African,” or “Asian,” say.  If we rearrange Eq. <a href="Week1.html#eq:1-1">(2.1)</a> and use the fact that the columns of <span class="math inline">\(\mathbf{U}\)</span> and <span class="math inline">\(\mathbf{V}\)</span> are <em>orthonormal</em>, we can find that
<span class="math display" id="eq:1-3">\[\begin{equation}
\mathbf{X}\mathbf{X}^T\mathbf{U}=\mathbf{U}\mathbf{\Sigma}^2,
\tag{2.3}
\end{equation}\]</span>meaning that the columns of <span class="math inline">\(\mathbf{U}\)</span> are the eigenvectors of the matrix
<span class="math display" id="eq:1-4">\[\begin{equation}
\frac{1}{m}\mathbf{X}\mathbf{X}^T
\tag{2.4}
\end{equation}\]</span> whose <span class="math inline">\(\left(i,j\right)\)</span> entry is the genetic correlation between individuals <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span>, sometimes known as the <em>genomic relationship matrix</em> or GRM.  PCA works by finding the first several eigenvectors of the GRM and plotting each individual’s ancestry along each orthogonal vector in a rectangular grid.  Clusters of individuals in this grid represent distinct ancestry groups.</p>
</div>
<div id="Week1_practice" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Principal components analysis: practice<a href="Week1.html#Week1_practice" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="Week1_practice_import" class="section level3 hasAnchor" number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> Importing the data<a href="Week1.html#Week1_practice_import" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To do PCA in R, we’ll need to load the libraries <kbd>SNPRelate</kbd> and <kbd>SeqArray</kbd>:</p>
<pre><code>library(SNPRelate)
library(SeqArray)</code></pre>
<p>Download the <a href="https://github.com/wletsou/BIOL-350/raw/master/docs/CHB%2BYRI%2BCEU.chr1.vcf.gz">chr1 vcf file</a> containing just the CEU, YRI, and CHB populations.  Once you have the file, store its name as a variable:</p>
<pre><code>vcf &lt;- &quot;path/to/file.vcf.gz&quot;</code></pre>
<p>Now we’ll convert the vcf format to gds format, retaining the base filename and changing the vcf.gz extension to gds.  (This may take a minute to complete.)  Then we’ll import the gds file as a gds object.</p>
<pre><code>seqVCF2GDS(vcf.fn = vcf,&quot;path/to/file.gds&quot;) # convert vcf to gds with a new file name
genofile &lt;- seqOpen(&quot;path/to/file.gds&quot;) # import the gds object</code></pre>
<p>You can can see the various fields under <kbd>genofile</kbd> by printing it.  To access the data in one of the fields, do</p>
<pre><code>seqGetData(genofile,&quot;sample.id&quot;) # view the sample ids</code></pre>
<p>where the name of the field is enclosed in quotes.</p>
</div>
<div id="Week1_practice_pca" class="section level3 hasAnchor" number="2.2.2">
<h3><span class="header-section-number">2.2.2</span> Running PCA<a href="Week1.html#Week1_practice_pca" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To run PCA in R, simply do</p>
<pre><code>pca &lt;- snpgdsPCA(genofile) # runs PCA</code></pre>
<p>to create an objects with 32 eignevectors of the GRM.  Make a data frame of the first several eigenvectors along with subject ids:</p>
<pre><code>df.pca &lt;- data.frame(sample = pca$sample.id,EV1 = pca$eigenvect[,1],EV2 = pca$eigenvect[,2],EV3 = pca$eigenvect[,3],stringsAsFactors = FALSE)</code></pre>
<p>We’ll plot individuals along EV1, EV2, and EV3 in several two-dimensional projections.  But we’ll want to see how the clustering done by PCA corresponds to individuals’ self-reported race; for that we’ll need another column in our data frame.</p>
</div>
<div id="Week1_practice_labels" class="section level3 hasAnchor" number="2.2.3">
<h3><span class="header-section-number">2.2.3</span> Getting the population labels<a href="Week1.html#Week1_practice_labels" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <a href="https://raw.githubusercontent.com/wletsou/BIOL-350/master/docs/CHB%2BYRI%2BCEU.txt">indivs file</a> contains each subject id along with its 1KGP population group.  Let’s import it now:</p>
<pre><code>indivs &lt;- read.table(&quot;path/to/CHB+YRI+CEU.txt&quot;,header = FALSE)
colnames(indivs) &lt;- c(&quot;id&quot;,&quot;pop&quot;)</code></pre>
<p>The second field of this table is <kbd>pop</kbd>, an assignment to each <kbd>id</kbd> of one of three 1KG population groups.  We want to match the right ID in <kbd>indivs</kbd> to the right ID in <kbd>df.pca</kbd> so that we can color our PCA plots by population.  If the tables are in the same order, matching will be easy, but it not, we have to use the <kbd>match(x,y)</kbd> function, which finds the positions in <kbd>y</kbd> corresponding to the same items in <kbd>x</kbd>.  Thus we can make a new column <kbd>pop</kbd> in <kbd>df.pca</kbd> with the corresponding <kbd>pop</kbd> values from <kbd>indivs</kbd> by</p>
<pre><code>df.pca$pop[match(indivs$id,df.pca$sample)] &lt;- indivs$pop # find the population group of each individual in df.pca</code></pre>
</div>
<div id="Week1_practice_plot" class="section level3 hasAnchor" number="2.2.4">
<h3><span class="header-section-number">2.2.4</span> Plotting<a href="Week1.html#Week1_practice_plot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now that we have a column of population labels, we can make a scatter plot colored by treating the <kbd>pop</kbd> column as vector of factors; we can get the unique values of a factor vector by applying the function <kbd>levels()</kbd> to it.  A plot of the second principal component vs. the first can then be generated by</p>
<pre><code>par(mar = c(5.1,5.1,4.1,2.1) ) 
plot(df.pca$EV1,df.pca$EV2,pch = 19,col = factor(df.pca$pop),xlab = &quot;PC1&quot;,ylab = &quot;PC2&quot;,cex.lab = 1.5,cex.axis = 1.5,cex.main = 1.5) # plot of PC2 vs. PC1
legend(&quot;topright&quot;,legend = levels(factor(df.pca$pop)),bty = &quot;n&quot;,pch = 19,col = factor(levels(factor(df.pca$pop))),pt.cex = 1,cex = 1.5,x.intersp = 0.2) # with a legend</code></pre>
<p>Move the legend around if it covers any points, and make similar plots for the other two comparisons between the first three PCs.</p>
<p>Finally, close the connection to the gds file when you are done:</p>
<pre><code>seqClose(genofile)</code></pre>
</div>
<div id="Week1_practice_turnin" class="section level3 hasAnchor" number="2.2.5">
<h3><span class="header-section-number">2.2.5</span> To turn in:<a href="Week1.html#Week1_practice_turnin" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Make three (nicely formatted) plots of:</p>
<ol style="list-style-type: decimal">
<li>PC2 vs. PC1</li>
<li>PC3 vs. PC1</li>
<li>PC3 vs. PC2</li>
</ol>
<p>For each plot, discuss:</p>
<ol style="list-style-type: decimal">
<li>Whether the populations appear to be well separated in PCA space</li>
<li>What the gradients of the different PCs represent, that is, what axis of variation each PC appears to explain</li>
<li>How to subset your <kbd>df.pca</kbd> data frame to isolate individuals of each population (i.e., provide code)</li>
</ol>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="Week2.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/wletsou/BIOL 350/edit/master/01-Week1.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["series.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
},
"citation_package": "biblatex"
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
