<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Kinship Analysis | BIOL 350: Bioinformatics</title>
  <meta name="description" content="Tutorial on genetic association studies." />
  <meta name="generator" content="bookdown 0.38 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Kinship Analysis | BIOL 350: Bioinformatics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Tutorial on genetic association studies." />
  <meta name="github-repo" content="openscapes/series" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Kinship Analysis | BIOL 350: Bioinformatics" />
  
  <meta name="twitter:description" content="Tutorial on genetic association studies." />
  

<meta name="author" content="William Letsou" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="Week1.html"/>
<link rel="next" href="Week3.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="lib/css/bootstrap.min.css" type="text/css" />
<link rel="stylesheet" href="lib/css/style.css" type="text/css" />
<link rel="stylesheet" href="lib/css/lesson.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">BIOL 350: Bioinformatics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="Week1.html"><a href="Week1.html"><i class="fa fa-check"></i><b>2</b> Principal Components Analysis</a>
<ul>
<li class="chapter" data-level="2.1" data-path="Week1.html"><a href="Week1.html#Week1_theory"><i class="fa fa-check"></i><b>2.1</b> Pricipal components analysis: theory</a></li>
<li class="chapter" data-level="2.2" data-path="Week1.html"><a href="Week1.html#Week1_practice"><i class="fa fa-check"></i><b>2.2</b> Principal components analysis: practice</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="Week1.html"><a href="Week1.html#Week1_practice_import"><i class="fa fa-check"></i><b>2.2.1</b> Importing the data</a></li>
<li class="chapter" data-level="2.2.2" data-path="Week1.html"><a href="Week1.html#Week1_practice_pca"><i class="fa fa-check"></i><b>2.2.2</b> Running PCA</a></li>
<li class="chapter" data-level="2.2.3" data-path="Week1.html"><a href="Week1.html#Week1_practice_labels"><i class="fa fa-check"></i><b>2.2.3</b> Getting the population labels</a></li>
<li class="chapter" data-level="2.2.4" data-path="Week1.html"><a href="Week1.html#Week1_practice_plot"><i class="fa fa-check"></i><b>2.2.4</b> Plotting</a></li>
<li class="chapter" data-level="2.2.5" data-path="Week1.html"><a href="Week1.html#Week1_practice_turnin"><i class="fa fa-check"></i><b>2.2.5</b> To turn in:</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="Week1.html"><a href="Week1.html#Week1_ref"><i class="fa fa-check"></i><b>2.3</b> References</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="Week2.html"><a href="Week2.html"><i class="fa fa-check"></i><b>3</b> Kinship Analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="Week2.html"><a href="Week2.html#Week2_theory"><i class="fa fa-check"></i><b>3.1</b> Kinship: theory</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="Week2.html"><a href="Week2.html#Week2_theory_king"><i class="fa fa-check"></i><b>3.1.1</b> KING</a></li>
<li class="chapter" data-level="3.1.2" data-path="Week2.html"><a href="Week2.html#Week2_theory_pcair"><i class="fa fa-check"></i><b>3.1.2</b> PC-AiR</a></li>
<li class="chapter" data-level="3.1.3" data-path="Week2.html"><a href="Week2.html#Week2_theory_pcrelate"><i class="fa fa-check"></i><b>3.1.3</b> PC-Relate</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="Week2.html"><a href="Week2.html#Week2_practice"><i class="fa fa-check"></i><b>3.2</b> Kinship: practice</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="Week2.html"><a href="Week2.html#Week2_practice_import"><i class="fa fa-check"></i><b>3.2.1</b> Importing data</a></li>
<li class="chapter" data-level="3.2.2" data-path="Week2.html"><a href="Week2.html#Week2_practice_pruning"><i class="fa fa-check"></i><b>3.2.2</b> PCA and LD-pruning</a></li>
<li class="chapter" data-level="3.2.3" data-path="Week2.html"><a href="Week2.html#Week2_practice_king"><i class="fa fa-check"></i><b>3.2.3</b> KING IBD and kinship calculation</a></li>
<li class="chapter" data-level="3.2.4" data-path="Week2.html"><a href="Week2.html#Week2_practice_pcair"><i class="fa fa-check"></i><b>3.2.4</b> PC-AiR</a></li>
<li class="chapter" data-level="3.2.5" data-path="Week2.html"><a href="Week2.html#Week2_practice_pcrelate"><i class="fa fa-check"></i><b>3.2.5</b> PC-Relate</a></li>
<li class="chapter" data-level="3.2.6" data-path="Week2.html"><a href="Week2.html#Week2_practice_turnin"><i class="fa fa-check"></i><b>3.2.6</b> To turn in:</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="Week2.html"><a href="Week2.html#Week2_ref"><i class="fa fa-check"></i><b>3.3</b> References</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="Week3.html"><a href="Week3.html"><i class="fa fa-check"></i><b>4</b> Association testing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="Week3.html"><a href="Week3.html#Week3_theory"><i class="fa fa-check"></i><b>4.1</b> Statistics</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="Week3.html"><a href="Week3.html#Week3_theory_or"><i class="fa fa-check"></i><b>4.1.1</b> The odds ratio</a></li>
<li class="chapter" data-level="4.1.2" data-path="Week3.html"><a href="Week3.html#Week3_theory_logit"><i class="fa fa-check"></i><b>4.1.2</b> The logistic model of disease risk</a></li>
<li class="chapter" data-level="4.1.3" data-path="Week3.html"><a href="Week3.html#Week3_theory_glm"><i class="fa fa-check"></i><b>4.1.3</b> Linear mixed models</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="Week3.html"><a href="Week3.html#Week3_practice"><i class="fa fa-check"></i><b>4.2</b> Practice</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="Week3.html"><a href="Week3.html#Week3_practice_pheno"><i class="fa fa-check"></i><b>4.2.1</b> Get phenotype</a></li>
<li class="chapter" data-level="4.2.2" data-path="Week3.html"><a href="Week3.html#Week3_practice_fit"><i class="fa fa-check"></i><b>4.2.2</b> Fitting the model</a></li>
<li class="chapter" data-level="4.2.3" data-path="Week3.html"><a href="Week3.html#Week3_practice_analysis"><i class="fa fa-check"></i><b>4.2.3</b> Analyzing the results</a></li>
<li class="chapter" data-level="4.2.4" data-path="Week3.html"><a href="Week3.html#to-turn-in-week3_practice_turnin"><i class="fa fa-check"></i><b>4.2.4</b> To turn in {#Week3_practice_turnin}:</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="Week3.html"><a href="Week3.html#Week3_ref"><i class="fa fa-check"></i><b>4.3</b> References</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">BIOL 350: Bioinformatics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="Week2" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Chapter 3</span> Kinship Analysis<a href="Week2.html#Week2" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="Week2_theory" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Kinship: theory<a href="Week2.html#Week2_theory" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Kinship can be defined as the expected fraction of alleles that two individuals got from the same ancestor(s).Â  We say that two individuals share an allele of a SNP <em>identical-by-descent</em> or <em>IBD</em> if they inherited the same copy of the allele from a common ancestor.Â  IBD-sharing is different from simply carrying the same allele of a gene (known as <em>identical-by-state</em> or <em>IBS</em>-sharing), which unrelated individuals may do if the allele is common enough in the population.Â  The <em>degree</em> <span class="math inline">\(R\)</span> of relationship may be defined as the effective number of meioses separating the relatives through the equation
<span class="math display" id="eq:2-1">\[\begin{equation}
\frac{1}{2^R}=\frac{1}{2^{R_1}}+\frac{1}{2^{R_2}},
\tag{3.1}
\end{equation}\]</span>
in which <span class="math inline">\(R_i\)</span> is the number of meioses separating the relatives through the first relativeâs <span class="math inline">\(i\)</span><sup>th</sup> parent.Â  For example, sibs are connected by two meioses through two parents, while a parent and child are connected by one meiosis through one parent: both relationships are degree-1.</p>
<p>The probability that two relatives share an allele IBD is <span class="math inline">\(\frac{1}{2^R}\)</span>, as there is a <span class="math inline">\(\frac{1}{2}\)</span> probability that an allele is passed on in any meiosis, and <span class="math inline">\(R\)</span> is the effective number of meioses or steps between the relatives.Â  If <span class="math inline">\(2\times\frac{1}{2^R}\)</span> is the expected number of alleles shared IBD at any given locus, then the fraction of the genome shared by any two relatives is
<span class="math display" id="eq:2-2">\[\begin{equation}
r=\frac{2\times\frac{1}{2^R}}{2}=\frac{1}{2^R}.
\tag{3.2}
\end{equation}\]</span></p>
<p>However, genomic sharing can be realized in different ways depending on the probabilities <span class="math inline">\(\pi_0\)</span>, <span class="math inline">\(\pi_1\)</span>, and <span class="math inline">\(\pi_2\)</span> that individuals share zero, one, or two copies IBD at a locus.Â  The probability that the relatives inherit both both copies IBD, viz.,
<span class="math display" id="eq:2-3">\[\begin{equation}
\pi_2=P\left(\text{share 2 IBD}\right)=2^2\frac{1}{2^{R_1}2^{R_2}},
\tag{3.3}
\end{equation}\]</span>is simply the product of the probabilities of sharing through both parents.Â  The probability of sharing exactly one allele IBD, viz.,
<span class="math display" id="eq:2-4">\[\begin{equation}
\pi_1=P\left(\text{share 1 IBD}\right)=2\left(\frac{1}{2^{R_1}}+\frac{1}{2^{R_2}}\right)-2^3\frac{1}{2^{R_1}2^{R_2}},
\tag{3.4}
\end{equation}\]</span>is the got by finding the probability <span class="math inline">\(2\left(\frac{1}{2^{R_1}}+\frac{1}{2^{R_2}}\right)-2^2\frac{1}{2^{R_1}2^{R_2}}\)</span> of sharing at least one allele IBD less the probability <span class="math inline">\(\pi_2\)</span> of sharing two.Â  Finally, the probability of sharing at least zero alleles IBD, viz.,
<span class="math display" id="eq:2-5">\[\begin{equation}
\pi_0=P\left(\text{share 0 IBD}\right)=1-2\frac{1}{2^{R_1}}-2\frac{1}{2^{R_2}}+2^2\frac{1}{2^{R_1}2^{R_2}},
\tag{3.5}
\end{equation}\]</span>is got by subtracting the probability <span class="math inline">\(\pi_1+\pi_2\)</span> of sharing at least one allele IBD from 1.Â  The coefficients account for the fact that there are <span class="math inline">\(2\)</span> alleles at each locus, and <span class="math inline">\(2^2\)</span> that can be shared.</p>
<p>From <a href="Week2.html#eq:2-3">(3.3)</a>â<a href="Week2.html#eq:2-5">(3.5)</a>, the fraction of the genome shared IBD is
<span class="math display" id="eq:2-4">\[\begin{equation}
r=\frac{2\pi_2+1\pi_1}{2}=\frac{1}{2^{R_1}}+\frac{1}{2^{R_2}}=\frac{1}{2^R}.
\tag{3.4}
\end{equation}\]</span>But the same value of <span class="math inline">\(r\)</span> can obtain from different values of <span class="math inline">\(\pi_1\)</span> and <span class="math inline">\(\pi_2\)</span>.Â  For example, full sibs have a 25% probability of sharing two alleles and a 50% chance of sharing one allele at a locus, for a total fraction <span class="math inline">\(r=0.5\)</span> shared.Â  But a parent and child have 0% chance of sharing two alleles and a 100% chance of sharing one, also giving <span class="math inline">\(r=0.5\)</span>.Â  Thus, your parent does is not equal your sibling, despite the fact of your sharing equal amounts of your genome with each of them.Â  Put another way, parents cannot pass on their genotypes to their offspring.</p>
<div id="Week2_theory_king" class="section level3 hasAnchor" number="3.1.1">
<h3><span class="header-section-number">3.1.1</span> KING<a href="Week2.html#Week2_theory_king" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>KING <span class="citation">[<a href="#ref-manichaikul_robust_2010">6</a>]</span> computes both the probability <span class="math inline">\(\pi_0\)</span> that two relatives share 0 alleles IBD as well as the coefficient of relatedness <span class="math inline">\(\phi=\frac{r}{2}\)</span>, defined as the probability that two alleles taken one from each relative are IBD at a locus (the maximum probability is <span class="math inline">\(\frac{1}{2}\)</span> because there is a 50% chance that the alleles chosen come from different parents).Â  The idea is to compare the counts <span class="math inline">\(X\)</span> and <span class="math inline">\(Y\)</span> of the alternative alleles which two individuals each have at a genetic locus.Â  If the pair are from a single ancestral population, the expected values and variances of the allele counts are <span class="math inline">\(\mathbb{E}\left(X\right)=\mathbb{E}\left(Y\right)=2p\)</span> and <span class="math inline">\(\sigma_X^2=\mathbb{E}\left(X^2\right)-\mathbb{E}\left(X\right)^2=\mathbb{E}\left(Y^2\right)-\mathbb{E}\left(Y\right)^2=2p\left(1-p\right)\)</span>.Â  Thus the expected value of the difference <span class="math inline">\(\mathbb{E}\left(\left(X-Y\right)^2\right)=\mathbb{E}\left(X^2\right)+\mathbb{E}\left(Y^2\right)-2\mathbb{E}\left(XY\right)\)</span> is
<span class="math display" id="eq:2-6">\[\begin{equation}\frac{\mathbb{E}\left(\left(X-Y\right)^2\right)}{\sigma_X^2+\sigma_Y^2}=1-\frac{\sigma_{XY}}{\sigma_X\sigma_Y}=1-r,
\tag{3.6}
\end{equation}\]</span>where <span class="math inline">\(\sigma_{XY}=\mathbb{E}\left(XY\right)-\mathbb{E}\left(X\right)\mathbb{E}\left(Y\right)\)</span> is the covariance of the genotype counts and <span class="math inline">\(r_{ij}=2\varphi_{ij}\)</span> is the genetic correlation between individuals <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span>; the latter can be interpreted as the amount of the genome shared IBD.</p>
<p>KING estimates <span class="math inline">\(\varphi\)</span> from Eq. <a href="Week2.html#eq:2-6">(3.6)</a> by counting the number <span class="math inline">\(N\)</span> of loci at which two individuals are heterozygous <span class="math inline">\(Aa,Aa\)</span> or opposite homozygous <span class="math inline">\(AA,aa\)</span>, as well as the total number of alleles at which each individual is heterozygous <span class="math inline">\(Aa\)</span>:
<span class="math display" id="eq:2-7">\[\begin{equation}
\hat{\varphi_{ij}}=\frac{N_{Aa,Aa}-2N_{AA,aa}}{N_{Aa}^{\left(i\right)}+N_{Aa}^{\left(j\right)}}.
\tag{3.7}
\end{equation}\]</span>From <a href="Week2.html#eq:2-7">(3.7)</a> it can be seen that shared heterozygous sites increase the estimated relatedness, and that unshared homozygous sites decrease relatedness.Â  Eq. <a href="Week2.html#eq:2-7">(3.7)</a> is called a ârobustâ estimator because it measures relatedness in a purely pairwise fashion: it does not rely on population estimates of allele frequencies.Â  However, if the individuals are not of the same genetic background, the allele frequency <span class="math inline">\(p\)</span> is not well-defined and Eq. <a href="Week2.html#eq:2-6">(3.6)</a> does not hold, leading to negative estimates of <span class="math inline">\(\phi\)</span>; this feature is not necessarily a problem, as it helps us to distinguish different ancestries within a single population.</p>
</div>
<div id="Week2_theory_pcair" class="section level3 hasAnchor" number="3.1.2">
<h3><span class="header-section-number">3.1.2</span> PC-AiR<a href="Week2.html#Week2_theory_pcair" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>KING provides an estimate of relatedness.Â  But as cryptic relatedness can skew ancestry estimates determined by PCA, we should use kinship to correct PCA.Â  This is where principal components analysis in related samples, or PC-AiR <span class="citation">[<a href="#ref-conomos_robust_2015">7</a>]</span> comes in handy.Â PC-AiR identifies a subset of <span class="math inline">\(\mathcal{U}\)</span> of individuals who are unrelated to each other (via negative KING kinship estimates), but maximally related to individuals in a remaining set <span class="math inline">\(\mathcal{R}\)</span>.Â Recalling the math in <a href="Week1.html#Week1">Week 1</a>, the matrix
<span class="math display" id="eq:2-8">\[\begin{equation}
\frac{1}{m}\mathbf{W}=\mathbf{X}^T\mathbf{U}=\mathbf{V}\mathbf{\Sigma},
\tag{3.8}
\end{equation}\]</span>so that <span class="math inline">\(\frac{1}{m}w_{jk}\lambda_{kk}^{-1}=x_{ij}u_{ik}\lambda_{kk}^{-1}=v_{jk}\)</span> is the genotype at SNP <span class="math inline">\(k\)</span> in ancestry <span class="math inline">\(j\)</span> inferred from subjectsâ genotypes <span class="math inline">\(\mathbb{X}\)</span> who are in subset <span class="math inline">\(\mathcal{U}\)</span>.Â  Then if we have an <span class="math inline">\(m\times n_u\)</span> genotype matrix <span class="math inline">\(\mathbf{X}&#39;\)</span> of individuals in the subset <span class="math inline">\(\mathcal{R}\)</span>, we can âapplyâ <span class="math inline">\(\mathbf{W}\)</span> to get an estimate of the âancestry-correctedâ PCs by:
<span class="math display" id="eq:2-9">\[\begin{equation}
\frac{1}{m}\mathbf{X}&#39;\mathbf{W}\mathbf{\Sigma}^{-1}=\mathbf{X}&#39;\mathbf{V}.
\tag{3.9}
\end{equation}\]</span>Eq. <a href="Week2.html#eq:2-9">(3.9)</a> is the matrix <span class="math inline">\(x&#39;_{i\cdot}v_{\cdot j}\)</span> of similarities between subjects <span class="math inline">\(i\)</span> in set <span class="math inline">\(\mathcal{R}\)</span> and ancestry <span class="math inline">\(j\)</span>.Â  In this way, we have estimated genotype principal components for all individuals (i.e., those in <span class="math inline">\(\mathcal{U}\)</span> and <span class="math inline">\(\mathcal{R}\)</span>) without the problem confounding due to cryptic relatedness.</p>
</div>
<div id="Week2_theory_pcrelate" class="section level3 hasAnchor" number="3.1.3">
<h3><span class="header-section-number">3.1.3</span> PC-Relate<a href="Week2.html#Week2_theory_pcrelate" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Once we have ancestry-corrected principal components for all individuals, we can obtain âcorrectedâ expected genotypes <span class="math inline">\(\mathbb{E}\left(X_{ik}\right)=2p_{ik}\)</span> of SNPs <span class="math inline">\(k\)</span> for populations of admixed individuals <span class="math inline">\(i\)</span> using a method known as PC-Relate <span class="citation">[<a href="#ref-conomos_model-free_2016">8</a>]</span>.Â  Because in populations with both shared genetic ancestry and cryptic relatedness, we can use the PC estimates <span class="math inline">\(v_{kj}\)</span> of the allele frequencies in each population <span class="math inline">\(j\)</span> to express the scaled genotypes <span class="math inline">\(x_{ij}=\frac{g_{ik}-2p_k}{2p_k\left(1-p_k\right)}\)</span> as
<span class="math display" id="eq:2-9">\[\begin{equation}
\mathbb{E}\left(x_{ik}\mid u_{ij}\right)=u_{ij}\lambda_{jj}v_{kj},
\tag{3.9}
\end{equation}\]</span>or, in terms of the measured genotypes:
<span class="math display" id="eq:2-10">\[\begin{equation}
\mathbb{E}\left(g_{ik}\mid u_{ij}\right)=2p_k+u_{ij}\lambda_{jj}v_{kj}\cdot 2p_k\left(1-p_k\right),
\tag{3.10}
\end{equation}\]</span>in which <span class="math inline">\(p_k\)</span> is the allele frequency estimated from the entire sample.Â  Eq. <a href="Week2.html#eq:2-10">(3.10)</a> says that if we make a graph of each individualâs genotype <span class="math inline">\(g_{ik}=0,1,2\)</span> at SNP <span class="math inline">\(k\)</span> vs.Â the (scaled) of the amount the individualâs population-<span class="math inline">\(j\)</span> genetic ancestry, the slope <span class="math inline">\(\lambda_{jj}v_{kj}2p_k\left(1-p_k\right)\)</span> should be the expected allele frequency in population <span class="math inline">\(j\)</span>.Â  In practice, the slopes are obtained from linear regression of the observed genotypes on genetic PCs, and the updated expected allele frequencies <span class="math inline">\(\mathbb{E}\left(g_{ik}\mid u_{ij}\right)=2p_{ik}\)</span> of SNPs <span class="math inline">\(k\)</span> for individuals <span class="math inline">\(i\)</span> are found by the corresponding value on the best-fit hyperplane, given individual <span class="math inline">\(i\)</span>âs genetic ancestry <span class="math inline">\(u_{ij}\)</span>.Â  We can then obtain a new genetic relationship matrix:
<span class="math display" id="eq:2-11">\[\begin{equation}
2\hat{\varphi}_{ij}=\frac{\sum_k\left(g_{ik}-2p_{ik}\right)\left(g_{jk}-2p_{jk}\right)}{2\sqrt{p_{ik}\left(1-p_{ik}\right)p_{jk}\left(1-p_{jk}\right)}}.
\tag{3.11}
\end{equation}\]</span>This new GRM will be used in <a href="Week3.html#Week3">Week 3</a> when we attempt to fit a linear mixed model for the effect of genotype on disease risk; for now we will use it to make an updated table of the relationship between individuals.</p>
</div>
</div>
<div id="Week2_practice" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Kinship: practice<a href="Week2.html#Week2_practice" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="Week2_practice_import" class="section level3 hasAnchor" number="3.2.1">
<h3><span class="header-section-number">3.2.1</span> Importing data<a href="Week2.html#Week2_practice_import" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now weâll try kinship analysis for ourselves using a simulated GWAS dataset.Â  This dataset was created using the program bioGWAS <span class="citation">[<a href="#ref-changalidis_biogwas_2023">9</a>]</span>, which uses other programs to simulate haplotypes <span class="citation">[<a href="#ref-su_hapgen2_2011">10</a>]</span> and phenotypes <span class="citation">[<a href="#ref-meyer_phenotypesimulator_2018">11</a>]</span> based on a set of input variants previously associated with breast cancer <span class="citation">[<a href="#ref-michailidou_genome-wide_2015">12</a>]</span> and made available by the MRC Interactive Epidemiology Unitâs Open GWAS Project <span class="citation">[<a href="#ref-elsworth_mrc_2020">13</a>]</span>.Â  This particular <a href="https://github.com/wletsou/BIOL-350/raw/master/docs/EUR_BCa.vcf.gz">vcf file</a> contains 1000 chromosome 10 genotypes at 345,130 variants (304,770 of which are SNPs) generated from 263 1KGP females of European origin.Â  Download and unzip the file:</p>
<pre><code>gunzip EUR_BCa.vcf.gz</code></pre>
<p>and move it to your desired location.Â  The unzipped file is readable in plain text format, although it is very big.</p>
<p>Next load the following R packages:</p>
<pre><code>library(gdsfmt)
library(GWASTools)
library(SNPRelate)
library(GENESIS)</code></pre>
<p>Our first task will be to import the vcf file and convert it to gds format.Â  First specify the directory in which you saved your <kbd>EUR_BCa.vcf</kbd>:</p>
<pre><code>directory &lt;- &quot;path/to/file/EUR_BCa.vcf # fill in the appropriate path</code></pre>
<p>Â  Then define two file names:</p>
<pre><code>vcf.fn &lt;- sprintf(&quot;%s/EUR_BCa.vcf&quot;,directory) # vcf file name (exists already)
gds.fn &lt;- sprintf(&quot;%s/EUR_BCa.gds&quot;,directory) # gds file name (about to be created)</code></pre>
<p>Next perform the conversion and importation into R:</p>
<pre><code>snpgdsVCF2GDS(vcf.fn,gds.fn) # create gds file from vcf
genofile &lt;- snpgdsOpen(gds.fn,readonly = FALSE) # import gds object</code></pre>
<p>The GDS format is an efficient representation of vcf files which does not require that the entire file be loaded into memory at once <span class="citation">[<a href="#ref-zheng_high-performance_2012">4</a>]</span>.Â  By typing</p>
<pre><code>genofile</code></pre>
<p>you can see all the fields under your gds object.Â  To access one of these nodes, type</p>
<pre><code>index.gdsn(genofile,&quot;sample.id&quot;) # vector of sample names</code></pre>
<p>or</p>
<pre><code>index.gdsn(genofile,&quot;genotype&quot;,start = c(1,1), count = c(10,10)) # 10 Ã 10 sample of the SNP genotype matrix</code></pre>
<p>This second line uses the <kbd>start</kbd> and <kbd>count</kbd> options to extract a range of data from the 1000 Ã 304,770 matrix of genotypes.Â  If you have a vector of phenotypes (as will will in <a href="Week3.html#Week3">Week 3</a>), you can add it as a node in your gds:</p>
<pre><code>add.gdsn(genofile,&quot;phenotype&quot;,val = fam$Phenotype) # add vector of phenotypes from a table</code></pre>
</div>
<div id="Week2_practice_pruning" class="section level3 hasAnchor" number="3.2.2">
<h3><span class="header-section-number">3.2.2</span> PCA and LD-pruning<a href="Week2.html#Week2_practice_pruning" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For now, we can run PCA on the genotype matrix by simply typing</p>
<pre><code>pca &lt;- snpgdsPCA(genofile) # principal components analysis</code></pre>
<p>We do not even have to specify the genotype node of <kbd>genofile</kbd>.Â  Make a plot of the first few PCs as we did in <a href="">Week 1</a>.</p>
<p>There are 304770 SNPs in this dataset; many are in LD with their neighbors.Â  To find a set of independent SNPs, which are not in LD (<span class="math inline">\(r&gt;0.10\)</span>) inside a sliding window of size ten million bp, we run:</p>
<pre><code>set.seed(566) # set random seed to initiate pruning process
snpset &lt;- snpgdsLDpruning(genofile,method = &quot;corr&quot;, slide.max.bp = 10e6,ld.threshold = sqrt(0.1), verbose = FALSE) # LD prining
pruned &lt;- unlist(snpset, use.names = FALSE) # set of independent SNPs</code></pre>
</div>
<div id="Week2_practice_king" class="section level3 hasAnchor" number="3.2.3">
<h3><span class="header-section-number">3.2.3</span> KING IBD and kinship calculation<a href="Week2.html#Week2_practice_king" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>With our set of pruned SNPs, we can run KING to compute the pairwise kinship coefficient (<span class="math inline">\(\varphi_{ij}=\frac{1}{2}r_{ij}\)</span>).Â  We will extract the <kbd>kinship</kbd> matrix from the created <kbd>ibd</kbd> object and rename the rows and columns with the ids of the samples:</p>
<pre><code>ibd &lt;- snpgdsIBDKING(genofile,snp.id = pruned) # KING kinship estimation
colnames(ibd$kinship) &lt;- ibd$sample.id
rownames(ibd$kinship) &lt;- ibd$sample.id</code></pre>
<p>Print a 10 Ã 10 sample of the <kbd>kinship</kbd> matrix using:</p>
<pre><code>ibd$kinship[1:10,1:10]</code></pre>
<p>Do the results make sense?Â  Hint: what should an individualâs kinship be with itself?</p>
<p>Finally, we will make a plot of kinship vs.Â the fraction of the genome shared IBS = 0.Â  This will take a minute or two to plot, because the data consist of all pairwise observations.</p>
<pre><code>par(mar = c(5.1,5.1,4.1,2.1) ) # left default plus one
plot(ibd$IBS0,ibd$kinship,ylab = &quot;Kinship coeffecient&quot;,xlab = &quot;IBS0&quot;,main = &quot;KING relatedness estimation&quot;, ylim = c(0,1),pch = 19,cex.lab = 1.5,cex.axis = 1.5,cex.main = 1.5) # plot kinship coefficient vs. proportion of SNPs not shared IBS</code></pre>
<p>You should notice that related individuals have a low fraction of IBS = 0, whereas unrelated individuals have a larger proportion.Â  The graph should be approximately linear, but with discontinuities as the relationsips become closer.Â  There should not be many close relatives in this dataset.Â  What, for example, is the largest inferred value of <span class="math inline">\(\varphi\)</span>?</p>
<p>In addition, if you remove the <kbd>ylim = c(0,1)</kbd> option, you will see a large cluster of negative kinship coefficients.Â  These values are expected if the (simulated) individuals are from different genetic backgrounds.</p>
<p>Before proceding to the next step, close your gds file:</p>
<pre><code>closefn.gds(genofile)</code></pre>
</div>
<div id="Week2_practice_pcair" class="section level3 hasAnchor" number="3.2.4">
<h3><span class="header-section-number">3.2.4</span> PC-AiR<a href="Week2.html#Week2_practice_pcair" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The PC-AiR method is part of the <kbd>GENESIS</kbd> package, which imports gds files through:</p>
<pre><code>geno &lt;- GdsGenotypeReader(sprintf(&quot;%s/pat_filt_sim.gds&quot;,directory)) # import the gds file you created above
genoData &lt;- GenotypeData(geno)</code></pre>
<p>and converts the objects to <kbd>GenotypeData</kbd> format <span class="citation">[<a href="#ref-gogarten_genetic_2019">14</a>]</span>.</p>
<p>Now we can run PC-Air.Â  It is important to include the kinship estimates we got from KING, as well as the list of pruned SNPs.Â  These values will help PC-AiR determine a set <span class="math inline">\(\mathcal{U}\)</span> of ancestry-representative individuals and their principal components, which will then imputed into the remaining subset <span class="math inline">\(\mathcal{R}\)</span> of individuals:</p>
<pre><code>mypcair &lt;- pcair(genoData,kinobj = ibd$kinship,divobj = ibd$kinship,snp.include = pruned) # genotype principal components based on a subset of unrelated individuals</code></pre>
<p>Next we can make a plot of the first few PCs using:</p>
<pre><code>par(mar = c(5.1,5.1,4.1,2.1) ) # left default plus one
plot(mypcair,vx = 1, vy = 2) # plot of PC2 vs PC1</code></pre>
<p>Change the values of <kbd>vx</kbd> and <kbd>vy</kbd> to change which PCs are plotted.</p>
</div>
<div id="Week2_practice_pcrelate" class="section level3 hasAnchor" number="3.2.5">
<h3><span class="header-section-number">3.2.5</span> PC-Relate<a href="Week2.html#Week2_practice_pcrelate" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>With our improved PCA results from PC-AiR, we will next update the GRM using individual-specific expected genotype counts <span class="math inline">\(2p_{ik}\)</span>.Â  First we need to create an interator that evaluates blocks of SNPs one at a time.</p>
<pre><code>genoData.iterator &lt;- GenotypeBlockIterator(genoData,snpInclude = pruned)</code></pre>
<p>From our pruned set of SNPs, you should have only one block.</p>
<p>Next we can run PC-Relate to update the kinship coefficients.Â  From the PCs determined from PC-Air, we can get the new GRM by running:</p>
<pre><code>mypcrel &lt;- pcrelate(genoData.iterator,pcs = mypcair$vectors[,1:2],training.set = mypcair$unrels) # kinship based on unrelated individuals</code></pre>
<p>The output object contains a field called <kbd>kinBtwn</kbd> which contains subfields <kbd>k0</kbd> and <kbd>kin</kbd> corresponding to the fraction of the genome shared IBD = 0 and the kinship between each pair of individuals.Â  Make a plot of these values, similar to KINGâs, using:</p>
<pre><code>par(mar = c(5.1,5.1,4.1,2.1) ) # left default plus one
plot(mypcrel$kinBtwn$k0,mypcrel$kinBtwn$kin,xlab = &quot;IBD0 proportion&quot;,ylab = &quot;Kinship coefficient&quot;,main = &quot;PC-relate relatedness estimation&quot;,pch = 19,cex.axis = 1.5,cex.lab = 1.5,cex.main = 1.5) # kinship vs. IBD0</code></pre>
<p>Finally, we convert the results of <kbd>pcrelate</kbd> to a GRM:</p>
<pre><code>myGRM &lt;- pcrelateToMatrix(mypcrel,scaleKin = 2) # genomic relationship matrix, multiplying each phi by 2</code></pre>
<p>We scale by a factor of 2 in this code so that each kinship is multiplied by 2: then the entries of the GRM will represent the fraction of the genome shared IBD.Â  Extract a 10 Ã 10 subset of this matrix to see if the results make sense, and how they compare to KINGâs kinship matrix computed above:</p>
<pre><code>myGRM[1:10,1:10]</code></pre>
<p>Now you can close the file:</p>
<pre><code>close(genoData)</code></pre>
<p>Save all your steps from todayâs analysisâwe will be repeating them in <a href="">Week 3</a> when we perform the association study.</p>
</div>
<div id="Week2_practice_turnin" class="section level3 hasAnchor" number="3.2.6">
<h3><span class="header-section-number">3.2.6</span> To turn in:<a href="Week2.html#Week2_practice_turnin" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Generate the following plots generated using the different methods, and comment on how they compare.</p>
<ol style="list-style-type: decimal">
<li>PCA plots of a) PC1 vs.Â PC2, b) PC1 vs.Â PC3, and c) PC2 vs.Â PC3 generated by both <kbd>snpgdsPCA</kbd> and PC-AiR.</li>
<li>Plots of kinship vs.Â IBD0 fraction generated by both KING and PC-Relate.</li>
<li>10 Ã 10 samples of the GRM generated by both KING and PC-Relate.</li>
</ol>
</div>
</div>
<div id="Week2_ref" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> References<a href="Week2.html#Week2_ref" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="refs" class="references csl-bib-body">
<div class="csl-entry">
1. The 1000 Genomes Project Consortium, Auton A, Abecasis GR, Steering committee, Altshuler DM, Durbin RM, et al. <a href="https://doi.org/10.1038/nature15393">A global reference for human genetic variation</a>. Nature. 2015;526:68â74.
</div>
<div class="csl-entry">
2. Price AL, Patterson NJ, Plenge RM, Weinblatt ME, Shadick NA, Reich D. <a href="https://doi.org/10.1038/ng1847">Principal components analysis corrects for stratification in genome-wide association studies</a>. Nature Genetics. 2006;38:904â9.
</div>
<div class="csl-entry">
3. Shlens J. <a href="http://arxiv.org/abs/1404.1100">A <span>Tutorial</span> on <span>Principal</span> <span>Component</span> <span>Analysis</span></a>. 2014.
</div>
<div class="csl-entry">
4. Zheng X, Levine D, Shen J, Gogarten SM, Laurie C, Weir BS. <a href="https://doi.org/10.1093/bioinformatics/bts606">A high-performance computing toolset for relatedness and principal component analysis of <span>SNP</span> data</a>. Bioinformatics. 2012;28:3326â8.
</div>
<div class="csl-entry">
5. Zheng X, Gogarten SM, Lawrence M, Stilp A, Conomos MP, Weir BS, et al. <a href="https://doi.org/10.1093/bioinformatics/btx145"><span>SeqArray</span>âa storage-efficient high-performance data format for <span>WGS</span> variant calls</a>. Bioinformatics. 2017;33:2251â7.
</div>
<div class="csl-entry">
6. Manichaikul A, Mychaleckyj JC, Rich SS, Daly K, Sale M, Chen W-M. <a href="https://doi.org/10.1093/bioinformatics/btq559">Robust relationship inference in genome-wide association studies</a>. Bioinformatics. 2010;26:2867â73.
</div>
<div class="csl-entry">
7. Conomos MP, Miller MB, Thornton TA. <a href="https://doi.org/10.1002/gepi.21896">Robust <span>Inference</span> of <span>Population</span> <span>Structure</span> for <span>Ancestry</span> <span>Prediction</span> and <span>Correction</span> of <span>Stratification</span> in the <span>Presence</span> of <span>Relatedness</span></a>. Genetic Epidemiology. 2015;39:276â93.
</div>
<div class="csl-entry">
8. Conomos MP, Reiner AP, Weir BS, Thornton TA. <a href="https://doi.org/10.1016/j.ajhg.2015.11.022">Model-free <span>Estimation</span> of <span>Recent</span> <span>Genetic</span> <span>Relatedness</span></a>. The American Journal of Human Genetics. 2016;98:127â48.
</div>
<div class="csl-entry">
9. Changalidis AI, Alexeev DA, Nasykhova YA, Glotov AS, Barbitoff YA. <a href="https://doi.org/10.3390/biology13010010"><span class="nocase">bioGWAS</span>: <span>A</span> <span>Simple</span> and <span>Flexible</span> <span>Tool</span> for <span>Simulating</span> <span>GWAS</span> <span>Datasets</span></a>. Biology. 2023;13:10.
</div>
<div class="csl-entry">
10. Su Z, Marchini J, Donnelly P. <a href="https://doi.org/10.1093/bioinformatics/btr341"><span>HAPGEN2</span>: Simulation of multiple disease <span>SNPs</span></a>. Bioinformatics. 2011;27:2304â5.
</div>
<div class="csl-entry">
11. Meyer HV, Birney E. <a href="https://doi.org/10.1093/bioinformatics/bty197"><span>PhenotypeSimulator</span>: <span>A</span> comprehensive framework for simulating multi-trait, multi-locus genotype to phenotype relationships</a>. Bioinformatics. 2018;34:2951â6.
</div>
<div class="csl-entry">
12. Michailidou K, BOCS, kConFab Investigators, AOCS Group, NBCS, GENICA Network, et al. <a href="https://doi.org/10.1038/ng.3242">Genome-wide association analysis of more than 120,000 individuals identifies 15 new susceptibility loci for breast cancer</a>. Nature Genetics. 2015;47:373â80.
</div>
<div class="csl-entry">
13. Elsworth B, Lyon M, Alexander T, Liu Y, Matthews P, Hallett J, et al. <a href="https://doi.org/10.1101/2020.08.10.244293">The <span>MRC</span> <span>IEU</span> <span>OpenGWAS</span> data infrastructure</a>. 2020.
</div>
<div class="csl-entry">
14. Gogarten SM, Sofer T, Chen H, Yu C, Brody JA, Thornton TA, et al. <a href="https://doi.org/10.1093/bioinformatics/btz567">Genetic association testing using the <span>GENESIS</span> <span>R</span>/<span>Bioconductor</span> package</a>. Bioinformatics. 2019;35:5346â8.
</div>
</div>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body">
<div id="ref-zheng_high-performance_2012" class="csl-entry">
4. Zheng X, Levine D, Shen J, Gogarten SM, Laurie C, Weir BS. <a href="https://doi.org/10.1093/bioinformatics/bts606">A high-performance computing toolset for relatedness and principal component analysis of <span>SNP</span> data</a>. Bioinformatics. 2012;28:3326â8.
</div>
<div id="ref-manichaikul_robust_2010" class="csl-entry">
6. Manichaikul A, Mychaleckyj JC, Rich SS, Daly K, Sale M, Chen W-M. <a href="https://doi.org/10.1093/bioinformatics/btq559">Robust relationship inference in genome-wide association studies</a>. Bioinformatics. 2010;26:2867â73.
</div>
<div id="ref-conomos_robust_2015" class="csl-entry">
7. Conomos MP, Miller MB, Thornton TA. <a href="https://doi.org/10.1002/gepi.21896">Robust <span>Inference</span> of <span>Population</span> <span>Structure</span> for <span>Ancestry</span> <span>Prediction</span> and <span>Correction</span> of <span>Stratification</span> in the <span>Presence</span> of <span>Relatedness</span></a>. Genetic Epidemiology. 2015;39:276â93.
</div>
<div id="ref-conomos_model-free_2016" class="csl-entry">
8. Conomos MP, Reiner AP, Weir BS, Thornton TA. <a href="https://doi.org/10.1016/j.ajhg.2015.11.022">Model-free <span>Estimation</span> of <span>Recent</span> <span>Genetic</span> <span>Relatedness</span></a>. The American Journal of Human Genetics. 2016;98:127â48.
</div>
<div id="ref-changalidis_biogwas_2023" class="csl-entry">
9. Changalidis AI, Alexeev DA, Nasykhova YA, Glotov AS, Barbitoff YA. <a href="https://doi.org/10.3390/biology13010010"><span class="nocase">bioGWAS</span>: <span>A</span> <span>Simple</span> and <span>Flexible</span> <span>Tool</span> for <span>Simulating</span> <span>GWAS</span> <span>Datasets</span></a>. Biology. 2023;13:10.
</div>
<div id="ref-su_hapgen2_2011" class="csl-entry">
10. Su Z, Marchini J, Donnelly P. <a href="https://doi.org/10.1093/bioinformatics/btr341"><span>HAPGEN2</span>: Simulation of multiple disease <span>SNPs</span></a>. Bioinformatics. 2011;27:2304â5.
</div>
<div id="ref-meyer_phenotypesimulator_2018" class="csl-entry">
11. Meyer HV, Birney E. <a href="https://doi.org/10.1093/bioinformatics/bty197"><span>PhenotypeSimulator</span>: <span>A</span> comprehensive framework for simulating multi-trait, multi-locus genotype to phenotype relationships</a>. Bioinformatics. 2018;34:2951â6.
</div>
<div id="ref-michailidou_genome-wide_2015" class="csl-entry">
12. Michailidou K, BOCS, kConFab Investigators, AOCS Group, NBCS, GENICA Network, et al. <a href="https://doi.org/10.1038/ng.3242">Genome-wide association analysis of more than 120,000 individuals identifies 15 new susceptibility loci for breast cancer</a>. Nature Genetics. 2015;47:373â80.
</div>
<div id="ref-elsworth_mrc_2020" class="csl-entry">
13. Elsworth B, Lyon M, Alexander T, Liu Y, Matthews P, Hallett J, et al. <a href="https://doi.org/10.1101/2020.08.10.244293">The <span>MRC</span> <span>IEU</span> <span>OpenGWAS</span> data infrastructure</a>. 2020.
</div>
<div id="ref-gogarten_genetic_2019" class="csl-entry">
14. Gogarten SM, Sofer T, Chen H, Yu C, Brody JA, Thornton TA, et al. <a href="https://doi.org/10.1093/bioinformatics/btz567">Genetic association testing using the <span>GENESIS</span> <span>R</span>/<span>Bioconductor</span> package</a>. Bioinformatics. 2019;35:5346â8.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="Week1.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="Week3.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/wletsou/BIOL 350/edit/master/02-Week2.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["series.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
},
"citation_package": "biblatex"
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
